### PR TITLE
Read joint rotation offsets from .fst

### PR DESCRIPTION
**Pull request by luiscuenca**
One requirement for the Improved Avatar Pipeline is to allow importing avatars with arbitrary skeleton rotations. These rotation offsets will be computed using existing external tools and the values added to the avatar .fst file.
This PR reads the avatar's joint rotation offsets contained on the .fst file and store them on the Rig class for further use.
https://highfidelity.manuscript.com/f/cases/19420/Implement-Skeleton-Reorienting-Feature-FST-Parsing

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14526/HighFidelity-Beta-PR14307-c070bce.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14526/HighFidelity-Beta-PR14307-c070bce.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14526/HighFidelity-Beta-Interface-PR14307-c070bce.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14526/HighFidelity-Beta-PR14307-c070bce.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14526/HighFidelity-Beta-Interface-PR14307-c070bce.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14537/HighFidelity-Beta-PR14307-914491a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14537/HighFidelity-Beta-PR14307-914491a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14537/HighFidelity-Beta-Interface-PR14307-914491a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14537/HighFidelity-Beta-PR14307-914491a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14537/HighFidelity-Beta-Interface-PR14307-914491a.exe

File: libraries/fbx/src/FSTReader.cpp [hunk 1]
```diff
@@ -207,6 +207,30 @@ QVector<QString> FSTReader::getScripts(const QUrl& url, const QVariantHash& mapp
...
+            QString jointName = itr.key();
+            QString line = itr.value().toString();
+            auto eulerAngles = line.split(',');
+            if (eulerAngles.size() == 3) {
+                float eulerX = eulerAngles[0].mid(1).toFloat();
+                float eulerY = eulerAngles[1].toFloat();
+                float eulerZ = eulerAngles[2].mid(0, eulerAngles[2].size() - 1).toFloat();
+                if (!isNaN(eulerX) && !isNaN(eulerY) && !isNaN(eulerZ)) {
+                    glm::quat rotationOffset = (glm::angleAxis(eulerX * RADIANS_PER_DEGREE, Vectors::UNIT_X) *
+                        glm::angleAxis(eulerY * RADIANS_PER_DEGREE, Vectors::UNIT_Y) *
```
Comment (amantley) [inline libraries/fbx/src/FSTReader.cpp]: Just putting a note here so we don't forget to confirm the euler ordering coming from unity.  I think it is Z Y X.  Is so we need to change the order of transforms here.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14728/HighFidelity-Beta-PR14307-bcde016.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14728/HighFidelity-Beta-PR14307-bcde016.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14728/HighFidelity-Beta-Interface-PR14307-bcde016.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14728/HighFidelity-Beta-PR14307-bcde016.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14728/HighFidelity-Beta-Interface-PR14307-bcde016.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14735/HighFidelity-Beta-PR14307-8310dd0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14735/HighFidelity-Beta-PR14307-8310dd0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14735/HighFidelity-Beta-Interface-PR14307-8310dd0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14735/HighFidelity-Beta-PR14307-8310dd0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14735/HighFidelity-Beta-Interface-PR14307-8310dd0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14778/HighFidelity-Beta-PR14307-e0d1ef3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14778/HighFidelity-Beta-PR14307-e0d1ef3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14778/HighFidelity-Beta-Interface-PR14307-e0d1ef3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14778/HighFidelity-Beta-PR14307-e0d1ef3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14778/HighFidelity-Beta-Interface-PR14307-e0d1ef3.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14975/HighFidelity-Beta-PR14307-2586811.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14975/HighFidelity-Beta-PR14307-2586811.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14975/HighFidelity-Beta-Interface-PR14307-2586811.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14975/HighFidelity-Beta-PR14307-2586811.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14975/HighFidelity-Beta-Interface-PR14307-2586811.dmg

**Comment by hyperlogic** [comment]
This PR code will be included into a different PR. specifically for ticket https://highfidelity.manuscript.com/f/cases/19421/Implement-Skeleton-Reorienting-Feature-Animation

**Pull request closed without merging**